src/components/createsecret.vue

```diff
@@ -0,0 +1,297 @@
+import { projectGroupSettingsLink, projectSettingsLink } from '../util/link';
+import { PropType, Ref, computed, onMounted, ref, toRefs } from 'vue';
+import { useRouter } from 'vue-router';
+import { useAsyncState } from '@vueuse/core';
+import { useAppState } from '../app/appstate';
+import _ from 'lodash';
+
+interface SecretValues {
+  key: string;
+  value: string;
+}
+
+type FormError = Record<string, string | undefined>;
+
+export default {
+  name: 'createsecret',
+  props: {
+    ownertype: {
+      type: String,
+      required: true,
+    },
+    ownername: {
+      type: String,
+      required: true,
+    },
+    projectref: { type: Array as PropType<Array<string>>, required: true },
+    projectgroupref: { type: Array as PropType<Array<string>>, required: true },
+    allSecrets: {
+      type: Array as PropType<Array<SecretResponse>>,
+      required: true,
+    },
+    refType: { type: String, required: true },
+  },
+  setup(props) {
+    const { ownertype, ownername, projectref, projectgroupref, refType } =
+      toRefs(props);
+    const router = useRouter();
+    const api = useAPI();
+    const appState = useAppState();
+    const secretName = ref('');
+    const secretvalues = ref<SecretValues[]>([{ key: '', value: '' }]);
+    const createSecretError: Ref<unknown | undefined> = ref();
+    const secretValueErrors = ref<FormError[]>([]);
+    const fetchAbort = new AbortController();
+    const secretNamePattern = /^[a-zA-Z][a-zA-Z0-9]*([-]?[a-zA-Z0-9]+)+$/;
+    const secretNameError = ref<string | null>(null);
+    const keyErrors = ref<string[]>([]);
+
+    onMounted(() => {
+      refreshSecrets();
+    });
+
+    const apiProjectGroupRef = computed(() => {
+      if (refType.value === 'project') {
+        return [ownertype.value, ownername.value, ...projectref.value].join(
+          '/'
+        );
+      }
+      if (refType.value === 'projectgroup') {
+        return [
+          ownertype.value,
+          ownername.value,
+          ...projectgroupref.value,
+        ].join('/');
+      }
+      return '';
+    });
+
+    const fetchSecrets = async () => {
+      try {
+        return await api.getSecrets(
+          refType.value,
+          apiProjectGroupRef.value,
+          false,
+          fetchAbort.signal
+        );
+      } catch (e) {
+        if (e instanceof ApiError) {
+          if (e.aborted) return;
+        }
+        appState.setGlobalError(e);
+      }
+    };
+
+    const { state: secrets, execute: refreshSecrets } = useAsyncState(
+      async () => {
+        return await fetchSecrets();
+      },
+      undefined,
+      { immediate: false }
+    );
+    const validateSecretName = () => {
+      secretNameError.value = null;
+      if (!secretName.value.trim()) {
+        secretNameError.value = 'Secret name is required';
+      } else if (!secretNamePattern.test(secretName.value)) {
+        secretNameError.value = 'Secret name does not match the pattern';
```

Comment:
which pattern? There should be something explaining the secret name rules in an human readable form.